libs/components/flyout/testing/src/modules/flyout/flyout-harness.ts

```diff
@@ -0,0 +1,140 @@
+   */
+  public async clickPreviousIteratorButton(): Promise<void> {
+    const iterator = await this.#getIterator();
+
+    await iterator.clickPreviousButton();
+  }
+
+  /**
+   * Clicks the flyout's primary action button.
+   */
+  public async clickPrimaryActionButton(): Promise<void> {
+    const primaryAction = await this.#getPrimaryAction();
+
+    await primaryAction.click();
+  }
+
+  /**
+   * Clicks the flyout's close button.
+   */
+  public async closeFlyout(): Promise<void> {
+    const closeButton = await this.locatorFor('.sky-flyout-btn-close')();
+    await closeButton.click();
+  }
+
+  /**
+   * Gets the flyout's ARIA-describedby attribute.
+   */
+  public async getAriaDescribedBy(): Promise<string | null> {
+    return await (await this.#getFlyout()).getAttribute('aria-describedby');
+  }
+
+  /**
+   * Gets the flyout's ARIA-label attribute.
+   */
+  public async getAriaLabel(): Promise<string | null> {
+    return await (await this.#getFlyout()).getAttribute('aria-label');
+  }
+
+  /**
+   * Gets the flyout's ARIA-labelledby attribute.
+   */
+  public async getAriaLabelledBy(): Promise<string | null> {
+    return await (await this.#getFlyout()).getAttribute('aria-labelledby');
+  }
+
+  /**
+   * Gets the flyout's ARIA role.
+   */
+  public async getAriaRole(): Promise<string | null> {
+    return await (await this.#getFlyout()).getAttribute('role');
+  }
+
+  /**
+   * Gets the label for the flyout's permalink button.
+   */
+  public async getPermalinkButtonLabel(): Promise<string> {
+    const permalink = await this.#getPermalink();
+
+    return await permalink.text();
+  }
+
+  /**
+   * Gets the route for the flyout's permalink button.
+   */
+  public async getPermalinkButtonRoute(): Promise<string | null> {
+    const permalink = await this.#getPermalink();
+
+    return await permalink.getAttribute('href');
+  }
+
+  /**
+   * Gets the label for the flyout's primary action button.
+   */
+  public async getPrimaryActionButtonLabel(): Promise<string> {
+    const primaryAction = await this.#getPrimaryAction();
+
+    return await primaryAction.text();
+  }
+
+  async #getIterator(): Promise<SkyFlyoutIteratorHarness> {
+    const iterator = await this.locatorForOptional(SkyFlyoutIteratorHarness)();
+
+    if (iterator === null) {
+      throw new Error('Could not find iterator buttons.');
+    }
+
+    return iterator;
+  }
+
+  async #getPermalink(): Promise<TestElement> {
+    const permalink = await this.locatorForOptional(
+      '.sky-flyout-btn-permalink',
+    )();
+
+    if (permalink === null) {
+      throw new Error('Could not find permalink button.');
+    }
+
+    return permalink;
+  }
+
+  async #getPrimaryAction(): Promise<TestElement> {
+    const primaryAction = await this.locatorForOptional(
+      '.sky-flyout-btn-primary-action',
+    )();
+
+    if (primaryAction === null) {
+      throw new Error('Could not find primary action button.');
+    }
+
+    return primaryAction;
+  }
+}
```

Comment:
Can we give consumers a way to get the current flyout width? Also the min and max width. These values seem to be reflected in the `.sky-flyout-resize-handle`

Reply:
added